scipy/stats/_qmc.py

```diff
@@ -886,7 +886,7 @@ def random(self, n: IntNumber = 1) -> np.ndarray:
         if self.centered:
             samples = 0.5
```

Comment:
Maybe this-needs to import `Union`? (I am on my phone sorry I cannot really easily test).
```suggestion
             samples: Union[np.ndarray, float] = 0.5
```

Reply:
Explicitly declaring the type as a `Union` would be the proper way to fix this issue (or better yet: use the `|` operator).

Ideally mypy would do this automatically if a variable is assigned different types within a single collection of `if/elif/else` statements, but alas...

Reply:
Thanks @BvB93 . I updated my suggestion to use the operator (just not sure about the spacing here).

Reply:
I'd just stuck to the standard pep8 spacing rules here.

Reply:
That fixed it, thanks both!